src/frontend/src/pages/GanttPage/GanttPageWrapper.tsx

```diff
@@ -7,71 +7,202 @@ import LoadingIndicator from '../../components/LoadingIndicator';
 import { useAllProjects } from '../../hooks/projects.hooks';
 import ErrorPage from '../ErrorPage';
 import { Task } from './GanttPackage/types/public-types';
-import { useAllWorkPackages } from '../../hooks/work-packages.hooks';
 import PageTitle from '../../layouts/PageTitle/PageTitle';
 import { Project, WbsElementStatus, WorkPackage } from 'shared';
 import GanttPage from './GanttPage';
 import { projectWbsPipe, wbsPipe } from '../../utils/pipes';
+import GanttPageFilter from './GanttPageFilter';
+import { ChangeEvent, FC, useEffect, useMemo, useState } from 'react';
+import { SelectChangeEvent } from '@mui/material/Select';
+import { useQuery } from '../../hooks/utils.hooks';
+import { useHistory } from 'react-router-dom';
+
+export const filterGanttProjects = (
```

Comment:
i think functions like this could go in a separate file like gantt.ts like we have form.ts